src/mlio/default_instance_reader.cxx

```diff
@@ -158,20 +161,58 @@ default_instance_reader::handle_nested_errors()
 std::optional<memory_slice>
 default_instance_reader::read_record_payload()
 {
+    std::string split_record_err_str = "Split record is not formatted with the correct cflag.";
```

Comment:
You should also give more context about the error here. See the `handle_nested_errors()` function where we include at least the data store's name in the error message.

Reply:
This gets caught by the handle nested error, which returns back the record index and the data store.